backend/src/main/java/de/bund/digitalservice/ris/norms/domain/entity/Proprietary.java

```diff
@@ -195,6 +195,22 @@ public void setRessort(final String newValue, final LocalDate date) {
       );
   }
 
+  /**
+   * Gets the date the expression enters into force
+   * @return the date of the entry into force
+   */
+  public Optional<LocalDate> getInkrafttreteDatum() {
```

Comment:
Not sure if these two methods belong to out domain model. I know reusing code for getting the metadata value but does this logic belong to our domain which represents the LDML.de standard? I understood that we filter this way because migration is putting that data there but later on if we need such a logic of filtering for another reason, will be also accessing the RIS metadata? Or rather using the lifecycle?

Reply:
In my understanding the custom metadata fields are also part of our domain. Also if I remember correctly the plan is to move the metadata from the neuris-specific block to the standard fields once they are in the spec.

So once the lifecycle data is garanteed to be correct we should be able to "just" change these 2 methods to load the dates from the lifecycle and no longer need those metadata fields to get the correct dates. Now that I'm thinking about it these methods might actually be better placed in either the metadata class or even the norm class 🤔

Reply:
I am just asking myself why are we using those migration metadata to filter out data and not just using the lifecycle events? Didn't fully check it but isn't the lifecycle containing also that information and is exactly for that? But if it is not the case which would surprise me, yes lets keep this logic in our domain model

Reply:
it seems that the data is available in the lifecycle data as well (atleast for the two norms I checked)

`refersTo="ausserkrafttreten" source="attributsemantik-noch-undefiniert" type="generation"`
and
`refersTo="inkrafttreten" source="attributsemantik-noch-undefiniert" type="generation"`

As the ticket mentions the two metadata fields we used here I'll check with Christian if it is ok if we use the lifecycle data instead :)

Reply:
we will for now use the `ris:` metadata fields as they might handle some edgecases.